### ISSUE TITLE
Inconsistent use of default values in Options dictionaries

### ISSUE THREAD

**Issue by rtoy** [WebAudio/web-audio-api#997]
We are inconsistent in whether default values are specified in the option dictionaries for each node. It was initially done this way because I didn't want to have two places specifying the default, but perhaps we should reconsider this and add default values for all dictionaries members that have defaults.

It's unfortunate to have two places, but I think dictionaries make it much clearer and is the future.

**Comment by rtoy** [comment]
I think doing this will allow specifying the AudioNode constructors (#986) easier.  Then we can just say `new FooNode(c, options)` is basically equivalent to
```
n = c.createFoo()
n.attribute1 = options.attribute1
n.attribute2 = options.attribute2
n.param1.value = options.param1
n.param2.value = options.param2
...
```
This takes care of most nodes except for the ones where `createFoo` has arguments.

**Comment by rtoy** [comment]
This was fixed by #1016
